contracts/Agreement.sol

```diff
@@ -45,6 +47,7 @@ contract Agreement is AccessControl, Ownable {
     }
 
     function optInToDAO() public onlyMember {
+      require(contractActive, "The contract no longer allows opt-ins");
```

Comment:
This `require` declaration should also be included in the `optOutOfDAO` function

Reply:
On a side note, it's good practice to move this to a function modifier if it is used in muliple functions:
```
modifier onlyActive() {
    require(contractActive, "The contract no longer allows opt-ins");
    _;
  }

function optInToDAO() public onlyActive onlyMember {
 ....
}

```